src/main/java/com/ning/http/client/providers/grizzly/ConnectionManager.java

```diff
@@ -214,16 +219,38 @@ boolean isReadyInPool(final Connection c) {
         final ConnectionInfo<SocketAddress> ci = pool.getConnectionInfo(c);
         return ci != null && ci.isReady();
     }
-    
+
     static boolean isKeepAlive(final Connection connection) {
         return !IS_NOT_KEEP_ALIVE.isSet(connection);
     }
-    
+
     private static String getPartitionId(InetAddress overrideAddress, Request request,
             ProxyServer proxyServer) {
-        return (overrideAddress != null ? overrideAddress.toString() + "_" : "") + 
-            request.getConnectionPoolPartitioning()
-             .getPartitionKey(request.getUri(), proxyServer).toString();
+        String partitionId = (overrideAddress != null ? overrideAddress.toString() + "_" : "") +
+                   request.getConnectionPoolPartitioning()
+                           .getPartitionKey(request.getUri(), proxyServer).toString();
+
+        // In the NTLM case, add in the partitionId the authentication credentials, since one
```

Comment:
add in the partitionId the authentication credentials -> add the authenticatin credentials in the partitionId